pkg/ans/ans_test.go

```diff
@@ -0,0 +1,249 @@
+			},
+			wantErr: false,
+		},
+		{
+			name:           "Faulty JSON yields error",
+			serviceKeyJSON: `bli-da-blup`,
+			wantErr:        true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotAnsServiceKey, err := UnmarshallServiceKeyJSON(tt.serviceKeyJSON)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("UnmarshallServiceKeyJSON() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			assert.Equal(t, tt.wantAnsServiceKey, gotAnsServiceKey, "Got the wrong ans serviceKey")
+		})
+	}
+}
+
+func TestTranslateLogrusLogLevel(t *testing.T) {
+	tests := []struct {
+		name         string
+		level        logrus.Level
+		wantSeverity string
+		wantCategory string
+	}{
+		{
+			name:         "InfoLevel yields INFO and NOTIFICATION",
+			level:        logrus.InfoLevel,
+			wantSeverity: infoSeverity,
+			wantCategory: notificationCategory,
+		},
+		{
+			name:         "DebugLevel yields INFO and NOTIFICATION",
+			level:        logrus.DebugLevel,
+			wantSeverity: infoSeverity,
+			wantCategory: notificationCategory,
+		},
+		{
+			name:         "WarnLevel yields WARNING and ALERT",
+			level:        logrus.WarnLevel,
+			wantSeverity: warningSeverity,
+			wantCategory: alertCategory,
+		},
+		{
+			name:         "ErrorLevel yields ERROR and EXCEPTION",
+			level:        logrus.ErrorLevel,
+			wantSeverity: errorSeverity,
+			wantCategory: exceptionCategory,
+		},
+		{
+			name:         "FatalLevel yields FATAL and EXCEPTION",
+			level:        logrus.FatalLevel,
+			wantSeverity: fatalSeverity,
+			wantCategory: exceptionCategory,
+		},
+		{
+			name:         "PanicLevel yields FATAL and EXCEPTION",
+			level:        logrus.PanicLevel,
+			wantSeverity: fatalSeverity,
+			wantCategory: exceptionCategory,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotSeverity, gotCategory := TranslateLogrusLogLevel(tt.level)
+			assert.Equal(t, tt.wantSeverity, gotSeverity, "Got wrong severity")
+			assert.Equal(t, tt.wantCategory, gotCategory, "Got wrong category")
+		})
+	}
+}
+
+func TestUnmarshallEventJSON(t *testing.T) {
+	tests := []struct {
+		name          string
+		eventJSON     string
+		existingEvent Event
+		wantEvent     Event
+		wantErr       bool
+	}{
+		{
+			name:      "Proper event JSON yields correct event",
+			eventJSON: `{"eventType": "my event","eventTimestamp":1647526655}`,
+			wantEvent: Event{
+				EventType:      "my event",
+				EventTimestamp: 1647526655,
+			},
+			wantErr: false,
+		},
+		{
+			name:      "Merging events includes all parts",
+			eventJSON: `{"eventType": "my event", "eventTimestamp": 1647526655, "tags": {"we": "were", "here": "first"}, "resource": {"resourceGroup": "blarp", "resourceName": "was changed"}}`,
+			existingEvent: Event{
+				EventType: "Bleep",
+				Subject:   "Bloop",
+				Tags:      map[string]interface{}{"Some": 1.0, "Additional": "a string", "Tags": true},
+				Resource: &Resource{
+					ResourceType: "blurp",
```

Comment:
While all these bleeps and bloops and blurps are fun, maybe they should be changed to something else for the merge into master? 😄